include/storage/io.hpp

```diff
@@ -15,6 +15,16 @@ namespace storage
 namespace io
 {
 
+inline std::size_t readPropertiesSize() { return 1; }
+
+template <typename PropertiesT>
+inline void readProperties(boost::filesystem::ifstream &properties_stream,
+                           PropertiesT properties[],
+                           std::size_t PropertiesSize)
+{
+    properties_stream.read(properties, PropertiesSize);
```

Comment:
We're planning to rename the generic `getTypeSize()` functions to something like `getTypeCount()` to be more accurate